cdisutils/excel.py

```diff
@@ -32,50 +32,120 @@ def combine_headers(headers):
+
+    Args:
+        ws (openpyxl.worksheet.worksheet.Worksheet): Worksheet to read from
+        num_headers (int): Number of headers in the work sheet
+
+    Return:
+        dict: Dictionary of lists, i.e. `{'key': [val1, val2]}`, `'key'` being
+            column header name and the value being the list of column values
+    """
+    sheet_data = {}
+    rows = ws.rows
+    header = read_headers(rows, num_headers)
+    if header is None:
+        return sheet_data
+    for row in rows:
+        for h, cell in zip(header, row):
+            value = (cell.value.strip()
+                     if isinstance(cell.value, str) else cell.value)
+            if h not in sheet_data:
+                sheet_data[h] = [value]
+            elif value:
+                sheet_data[h].append(value)
+
+    return sheet_data
+
+
+def read_sheet(ws=None, num_headers=1):
+    """Read an XLSX spreadsheet and return a list of rows
+
+    Args:
+        ws (openpyxl.worksheet.worksheet.Worksheet): Worksheet to read from
+        num_headers (int): Number of headers in the worksheet. NOTE: maximum
+            allowed number of headers is `2`
+
+    Return:
+        list: List of rows in format:
+
+            sheet_data = [
+                {
+                    'header1': value11,
+                    'header2': value12
+                },
+                {
+                    'header1': value21,
+                    'header2': value22
+                }
+            ]
+    """
     sheet_data = []
-    cur_row = 1
-    header_combine = False
     header = None
     if num_headers > 2:
         log.info("Sorry, can only process 2 or fewer header rows")
-    else:
-        if num_headers > 1:
-            header_combine = True
-        log.info("{} rows in sheet".format(ws.max_row))
-
-        for row in ws.rows:
-            if num_headers > 0:
-                header_data = []
-                for cell in row:
-                    if cell.value:
-                        header_data.append(cell.value.strip())
-                    else:
-                        header_data.append(None)
-                headers.append(header_data)
-                num_headers -= 1
-                if num_headers == 0:
-                    if header_combine:
-                        header = combine_headers(headers)
-                    else:
-                        header = headers[0]
-            else:
-                # try and strip any blank rows
-                any_data = False
-                for cell in row:
-                    if cell.value:
-                        if len(cell.value.strip()):
-                            any_data = True
-                            break
-
-                if any_data:
-                    line_data = dict(zip(header, [cell.value for cell in row]))
-                    sheet_data.append(line_data)
-            cur_row += 1
+        return sheet_data
+
+    if num_headers > 1:
+        header_combine = True
```

Comment:
seems like `header` and `header_combine` can be safely removed since they are not used. Or even better can you make it look like L79-81

Reply:
@kulgan Done. Good catch. I completely missed this. Thanks!